tmux/tmux
# Add popup style arguments

This PR adds the `-s popup-style` and `-S popup-border-style` arguments to the `display-popup` command as suggested in [@nicm's comment on #2927](https://github.com/tmux/tmux/pull/2927#issuecomment-939905955):
> people will either want […] set them per-popup (which would mean they need to be added arguments to display-popup like `-B`

~~⚠️ The merge base of this PR should be [`afh:afh-popup-border-lines`](https://github.com/afh/tmux/tree/afh-popup-border-lines) from #2930 instead of [`tmux:master`](https://github.com/tmux/tmux).~~ Update: #2930 has been merged to master

~~❗ To see only the relevant changes [compare the `afh:afh-popup-border-lines` branch against the `afh:afh-popup-style-arguments` branch](https://github.com/afh/tmux/compare/afh-popup-border-lines...afh:afh-popup-style-arguments?expand=1).~~ Update this PR branch now only shows the relevant changes.

To test:
* Build and run tmux `./tmux -vv -f/dev/null -Ltest`
* hen test out the different popup borders (see `test_popup_arguments.sh ` and screenshots below for example commands)

<details><summary><code>test_popup_arguments.sh</code>: Script to test the <code>-s</code> and <code>-S</code> arguments of <code>display-popup</code></summary>

```
#!/bin/ksh

clear
echo "\$ ./tmux set -gw popup-style fg=white,bg=red"
./tmux set -gw popup-style fg=white,bg=red
echo "\$ ./tmux set -gw popup-border-style fg=red,bg=white"
./tmux set -gw popup-border-style fg=red,bg=white
echo "\$ ./tmux popup \"echo popup styled with options\""
./tmux popup echo "popup styled with options"
clear; echo "\$ ./tmux popup -s fg=white,bg=blue -S fg=magenta,bg=cyan \"echo
popup styled with arguments\""
./tmux popup -s fg=white,bg=blue -S fg=magenta,bg=cyan echo "popup styled with
arguments"
```

</details>

[tmux-client-63048.log](https://github.com/tmux/tmux/files/7341545/tmux-client-63048.log)
[tmux-out-63050.log](https://github.com/tmux/tmux/files/7341546/tmux-out-63050.log)
[tmux-server-63050.log](https://github.com/tmux/tmux/files/7341547/tmux-server-63050.log)

| Styled with Options | Styled with Arguments |
| --- | --- |
|![image](https://user-images.githubusercontent.com/16507/137215641-050bf6c5-40cd-4dfe-8995-7ad9f04c2836.png)|![image](https://user-images.githubusercontent.com/16507/137215647-900d3978-6feb-490b-b590-847c8c33a253.png)|
